code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -1205,6 +1205,79 @@
 
 	succeed_activate()
 
+//////////////////////////
+/// Build xeno jel turret
+//////////////////////////
+
+/datum/action/xeno_action/activable/build_turret_jel
+	name = "Secrete jel turret"
+	action_icon_state = "xeno_turret"
+	mechanics_text = "Creates a new xeno jel turret for 100 points"
+	ability_name = "secrete jel turret"
+	plasma_cost = 150
+	cooldown_timer = 60 SECONDS
+	gamemode_flags = ABILITY_DISTRESS
+	/// How long does it take to build
+	var/build_time = 15 SECONDS
+	/// Pyschic point cost
+	var/psych_cost = XENO_TURRET_PRICE
+
+/datum/action/xeno_action/activable/build_turret_jel/can_use_ability(atom/A, silent, override_flags)
```

Comment:
I'd suggest making this a subtype of the above action, `/datum/action/xeno_action/activable/build_turret`. That way you don't actually need to redefine this whole `can_use_ability()` proc and all of the variables, you can just override `use_ability()`.